### DISCUSSION TITLE
need Lua Guidance

### DISCUSSION THREAD

**Discussion by quimsical** [thatskymod/Sky-CotL-Scripts#156] [Q&A] [answered]
I have absolutely no experience with using a lua script. I first tried to just straight up copy and paste(60fps.lua) into a lua script executor app( the literal name is "exec lua" in play store) but came back with an error on the 10ᵗʰ line stating it didn't understand global "gg". i would very much would appreciate a (if possible in depth but basics really needed) tutorial on how to make the lua scripts work with the intended app.

**Comment by thatskymod** [comment, marked as answer]
The Lua script you are trying to run is designed specifically for use with GameGuardian, which is a tool used for modifying values in video games on mobile devices. The "gg" you're seeing refers to a special object in GameGuardian's scripting environment[^1] that provides functions such as searchNumber and refineNumber, which are not available in standard Lua environments. This is why your script is throwing an error in a standard Lua executor.

Nowadays, though, there's no real reason to deal with the complexities of GameGuardian since newer apps like Canvas offer mods in a more approachable way.

Still, GameGuardian remains a useful tool for those interested in advanced tweaking or low-level memory editing. It's best suited for professionals who already have some technical background.

[^1]: https://gameguardian.net/help/classgg.html
